presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/classloader/ClassLoaderSafeEventListener.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.base.classloader;
+
+import io.prestosql.spi.classloader.ThreadContextClassLoader;
+import io.prestosql.spi.eventlistener.EventListener;
+import io.prestosql.spi.eventlistener.QueryCompletedEvent;
+import io.prestosql.spi.eventlistener.QueryCreatedEvent;
+import io.prestosql.spi.eventlistener.SplitCompletedEvent;
+
+import static java.util.Objects.requireNonNull;
+
+public class ClassLoaderSafeEventListener
+        implements EventListener
+{
+    private final EventListener delegate;
+    private final ClassLoader classLoader;
+
+    public ClassLoaderSafeEventListener(EventListener delegate, ClassLoader classLoader)
```

Comment:
```suggestion
    public ClassLoaderSafeEventListener(@ForClassLoaderSafe  EventListener delegate, ClassLoader classLoader)
```

Reply:
What would that annotation do? This class is used as a decorator inside a stream/for-loop operation. Nothing will be injected into it through guice, as there are potentially many EventListeners etc. For me adding it there would be misleading, but maybe I don't get the full purpose.

Reply:
`ClassLoaderSafe*` classes are also designed to be used with guice. I wanted to add this annotation to make sure it follows the convention and behaves like any other `ClassLoaderSafe*` class.

Reply:
Please don't get me wrong. I don't want you to use it ;)

Reply:
ok I get it :)